doc/source/user_guide/pyarrow.rst

```diff
@@ -0,0 +1,141 @@
+.. _pyarrow:
+
+{{ header }}
+
+*********************
+PyArrow Functionality
+*********************
+
+pandas can utilize `PyArrow <https://arrow.apache.org/docs/python/index.html>`__ to extend functionality and improve the performance
+of various APIs. This includes:
+
+* More extensive `data types <https://arrow.apache.org/docs/python/api/datatypes.html>`__ compared to NumPy
+* Missing data support (NA) for all data types
+* Performant IO reader integration
+
+To use this functionality, please ensure you have :ref:`installed the minimum supported PyArrow version. <install.optional_dependencies>`
+
+
+Data Structure Integration
+--------------------------
+
+A :class:`Series`, :class:`Index`, or the columns of a :class:`DataFrame` can be directly backed by a :external+pyarrow:py:class:`pyarrow.ChunkedArray`
+which is similar to a NumPy array. To construct these from the main pandas data structures, you can pass in a string of the type followed by
+``[pyarrow]``, e.g. ``"int64[pyarrow]""`` into the ``dtype`` parameter
+
+.. ipython:: python
+
+   ser = pd.Series([-1.5, 0.2, None], dtype="float32[pyarrow]")
+   ser
+
+   idx = pd.Index([True, None], dtype="bool[pyarrow]")
+   idx
+
+   df = pd.DataFrame([[1, 2], [3, 4]], dtype="uint64[pyarrow]")
+   df
+
+For PyArrow types that accept parameters, you can pass in a PyArrow type with those parameters
+into :class:`ArrowDtype` to use in the ``dtype`` parameter.
+
+.. ipython:: python
+
+   import pyarrow as pa
+   list_str_type = pa.list_(pa.string())
+   ser = pd.Series([["hello"], ["there"]], dtype=pd.ArrowDtype(list_str_type))
+   ser
+
+   from datetime import time
+   idx = pd.Index([time(12, 30), None], dtype=pd.ArrowDtype(pa.time64("us")))
+   idx
+
+   from decimal import Decimal
+   decimal_type = pd.ArrowDtype(pa.decimal128(3, scale=2))
+   data = [[Decimal("3.19"), None], [None, Decimal("-1.23")]]
+   df = pd.DataFrame(data, dtype=decimal_type)
+   df
+
+If you already have an :external+pyarrow:py:class:`pyarrow.Array` or :external+pyarrow:py:class:`pyarrow.ChunkedArray`,
+you can pass it into :class:`.arrays.ArrowExtensionArray` to construct the associated :class:`Series`, :class:`Index:`
+or :class:`DataFrame` object.
+
+.. ipython:: python
+
+   pa_array = pa.array([{"1": "2"}, {"10": "20"}, None])
+   ser = pd.Series(pd.arrays.ArrowExtensionArray(pa_array))
+   ser
+
+
+Operations
+----------
+
+PyArrow data structure integration is implemented through pandas' :class:`~pandas.api.extensions.ExtensionArray` :ref:`interface <extending.extension-types>`;
+therefore, supported functionality exists where this interface is integrated within the pandas API. Additionally, this functionality
+is accelerated with PyArrow `compute functions <https://arrow.apache.org/docs/python/api/compute.html>`__ where available. This includes:
+
+* Numeric aggregations
+* Numeric arithmetic
+* Numeric rounding
+* Logical and comparison functions
+* String functionality
+* Datetime functionality
+
```

Comment:
Maybe worth adding that below are just some examples of native Arrow-backed functions? I feel a bit of context is missing to understand what you're trying to show with those examples.

Reply:
Good idea. Added a sentence that these are some examples that use PyArrow compute functions